kvserver: use wrapper type for Store.mu.replicas

tbg opened the pull request:
This simplifies lots of callers and it will also make it easier to work
on #72374, where this map will start containing more than one type as
value.

Release note: None

tbg committed:
kvserver: use wrapper type for Store.mu.replicas

This simplifies lots of callers and it will also make it easier to work
on #72374, where this map will start containing more than one type as
value.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/72430)
<!-- Reviewable:end -->

tbg committed:
kvserver: rename replicas to replicasByRangeID

Release note: None

erikgrinaker reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 7 of 7 files at r1, 1 of 6 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72430#-:-MnfIw148tz6_yRTwm8n:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @erikgrinaker and @tbg)

___
*[pkg/kv/kvserver/store_replicas_by_rangeid.go, line 23 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/72430#-MnfHdbiAaNd9gFEEkIq:-MnfHdbiAaNd9gFEEkIr:botqdq0) ([raw file](https://github.com/cockroachdb/cockroach/blob/6256dd84c561496f78f1f3724012177979d1bf8e/pkg/kv/kvserver/store_replicas_by_rangeid.go#L23)):*
> ```Go
> 
> type rangeIDReplicaMap struct {
> 	m syncutil.IntMap
> ```

nit: consider calling this `replicas` or something -- `m.m` is a bit terse.
___
*[pkg/kv/kvserver/store_replicas_by_rangeid.go, line 52 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/72430#-MnfIc5S4PKf5ZK1wAP4:-MnfIc5S4PKf5ZK1wAP5:bvq2imf) ([raw file](https://github.com/cockroachdb/cockroach/blob/6256dd84c561496f78f1f3724012177979d1bf8e/pkg/kv/kvserver/store_replicas_by_rangeid.go#L52)):*
> ```Go
> // returned from the closure to stop iteration without an error
> // resulting from Range().
> func (m *rangeIDReplicaMap) Range(f func(*Replica) error) error {
> ```

nit: looks like we're never returning an error from the closure, so I wonder if it'd be just as easy to return a bool here for consistency with `sync.Map` and avoid the errcheck appeasement.


<!-- Sent from Reviewable.io -->

tbg committed:
kvserver: touch up rangeIDReplicaMap

- avoid the struct wrapper (and thus the need to name the inner map)
- don't return an error in Range()

Release note: None

tbg commented:
TFTR! I addressed your comments.

bors r=erikgrinaker
